File: Level 2/귤 고르기/solution.js
function solution(k, tangerines) {
  const tangerineCountDataObj = {};
  let answer = 0;

  tangerines.forEach((tangerine) => {
    tangerineCountDataObj[tangerine] = tangerineCountDataObj[tangerine]
      ? tangerineCountDataObj[tangerine] + 1
      : 1;
  });

  const tangerineCountDataArray = Object.entries(tangerineCountDataObj).map(
    ([key, value]) => ({ key, value })
  );
  tangerineCountDataArray.sort((a, b) => b.value - a.value);

  tangerineCountDataArray.forEach(({ key, value }) => {
    if (k <= 0) {
      return;
    }

    k = k - value;
    answer++;
  });

  return answer;
}

File: Level 2/귤 고르기/solution.py
from collections import Counter


def solution(k, tangerines):
    counter = Counter(tangerines)

    sorted_counter = sorted(counter.values(), reverse=True)

    answer = 0

    for tangerine in sorted_counter:
        if k <= 0:
            return answer

        k -= tangerine
        answer += 1

    return answer
